views/Graph.react.js

```diff
@@ -390,54 +483,94 @@ class Graph extends Component {
     socket.close()
     window.removeEventListener('resize', this.handleResize)
   }
-
+  updateUserPreferences = (configuration) => {
+    let uid = this.props.user.uid
+    const createNewPreference = preparePreferences(
+      configuration,
+      this.state.preferences
+    )
+    return window
+      .fetch(`${basePath}/api/v1/users/${uid}/preferences`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        credentials: 'same-origin',
+        body: JSON.stringify({
+          preferences: createNewPreference,
+        }),
+      })
+      .then((res) => {
+        if (res.status === 201) {
+          window.location.replace(
+            routes.subjectView(
+              this.state.subject.project,
+              this.state.subject.sid
+            )
+          )
+        }
+      })
+  }
   render() {
-    const { loading, success } = this.state;
-    const { classes, theme } = this.props;
+    const { loading, success } = this.state
+    const { classes, theme } = this.props
     const buttonClassname = classNames({
       [classes.buttonSuccess]: success,
-    });
+    })
     return (
-      <div
-        className={classes.root}
-      >
+      <div className={classes.root}>
         <AppBar className={classes.appBar}>
           <Toolbar
             variant='dense'
             style={{
-              paddingLeft: '16px'
+              paddingLeft: '16px',
             }}
           >
             <IconButton
-              color="rgba(0, 0, 0, 0.54)"
-              aria-label="Open drawer"
+              color='rgba(0, 0, 0, 0.54)'
+              aria-label='Open drawer'
               onClick={this.handleDrawerToggle}
             >
-              <img width='24px' height='24px' src={`${basePath}/img/favicon.png`} />
+              <img
+                width='24px'
+                height='24px'
+                src={`${basePath}/img/favicon.png`}
+              />
             </IconButton>
             <Typography
-              variant="title"
-              color="inherit"
+              variant='title'
+              color='inherit'
               style={{
                 color: 'rgba(0,0,0,0.4)',
                 fontSize: '18px',
                 letterSpacing: '1.25px',
-                flexGrow: 1
+                flexGrow: 1,
               }}
             >
               {this.state.subject.sid + ' - ' + this.state.subject.project}
             </Typography>
+            <div className={classes.configDropDownContainer}>
+              <Typography className={classes.dropDownText}>
+                Configuration Settings
+              </Typography>
+              <ConfigDropDown
+                configurations={this.state.configurationsList}
+                updatePreferences={this.updateUserPreferences}
+                currentConfig={this.state.preferences}
```

Comment:
I'm confused by the naming of preferences and configurations. I thought they were different but this line makes me think maybe they are the same?

Reply:
They're different, configurations are uploaded configurations. 
The name is also tricky because it says preference, but there can only be one preference at a time from what i've learned. It should be just currentPreference or currentUserPreference and it has other data inside it.

Reply:
https://github.com/AMP-SCZ/dpdash/pull/129/commits/75fa4e051ac01363f4567f566d2f67ddf2280f34